### PR TITLE
Notify action though Azure Storage Queue

### PR DESCRIPTION
**Pull request by anastasiia-zolochevska**

### CODE REVIEW

File: tools/c7n_azure/c7n_azure/utils.py [hunk 1]
```diff
@@ -30,3 +31,9 @@ def get_resource_type(resource_id):
     @staticmethod
     def get_resource_name(resource_id):
         return resource_id.split('/')[8]
+
+
+def utcnow():
```
Comment (kapilt) [inline tools/c7n_azure/c7n_azure/utils.py]: a one line function for a one liner, feels a bit odd, though this is just a move.

Comment (stefangordon) [reply tools/c7n_azure/c7n_azure/utils.py]: To make patching/mocking utcnow() easier I'm guessing?

File: tools/c7n_azure/c7n_azure/actions.py [hunk 2]
```diff
@@ -345,3 +342,66 @@ def process_resource(self, resource):
             return
 
         self.untag_action({'tags': candidates}, self.manager).process([resource])
+
+
+class Notify(BaseNotify):
+
+    batch_size = 50
```
Comment (stefangordon) [inline tools/c7n_azure/c7n_azure/actions.py]: 14 bytes may be more wiggle room than we need? Hate to give up 21% of the message size.  Looks like they left a 6-byte buffer in the AWS implementation which makes me think 58 would work fine at least?

File: tools/c7n_azure/c7n_azure/utils.py (hunk 1 above)
Comment (stefangordon) [reply tools/c7n_azure/c7n_azure/utils.py]: oic, you just moved this - yes it is used in some unit tests where `now` needs patching - having a wrapper method makes it cleaner.

File: tools/c7n_azure/tests/test_notify.py [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+            'resource': 'azure.keyvault',
+            'actions': [
+                {'type': 'notify',
+                 'template': 'default',
+                 'priority_header': '2',
+                 'subject': 'testing notify action',
+                 'to': ['user@domain.com'],
+                 'transport':
+                     {'type': 'asq',
+                      'queue': 'https://cctstoragey6akyqpagdt3o.queue.core.windows.net/testcc'}
```
Comment (stefangordon) [inline tools/c7n_azure/tests/test_notify.py]: Won't be able to rerecord this as storage names are random each time.  You can copy how I did it in `tools/c7n_azure/tests/test_storageutils.py` with the `setup_account()` which goes and finds the storage account, e.g. 

```
account = self.setup_account()
url = "https://" + account.name + ".blob.core.windows.net/testcc"
```
You can also read your queue messages (and delete them), example in that same file, e.g.

```
        # Pull messages, should be 1
        messages = StorageUtilities.get_queue_messages(*queue_settings)
        self.assertEqual(len(messages), 1)

        # Read message and delete it from queue
        for message in messages:
            self.assertEqual(message.content, u"hello queue")
            StorageUtilities.delete_queue_message(*queue_settings, message=message)
```

File: tools/c7n_azure/c7n_azure/actions.py (hunk 2 above)
Comment (anastasiia-zolochevska) [reply tools/c7n_azure/c7n_azure/actions.py]: Batch size is actually not in KB, but in the number of resources. It’s a bit hard to predict how many resources we can batch together as resources differ, but 50 seems safe enough.

File: tools/c7n_azure/tests/test_notify.py (hunk 3 above)
Comment (anastasiia-zolochevska) [reply tools/c7n_azure/tests/test_notify.py]: Done. I moved setup_account method to the base class.

**Pull request merged**